### PR TITLE
Use qt6 by default

### LINKED ISSUE #3221: Support Qt6 + KSyntaxHighlighting
It seems that it supports Qt6 out of the box already: https://invent.kde.org/frameworks/syntax-highlighting/-/blob/master/CMakeLists.txt

It is just called `KF6SyntaxHighlighting` now

### PR DESCRIPTION
**Pull request by karliss**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**TODO** 

- [x] test and finish updating build setup instructions for various Linux distros
- [x] test and update paths for windows build instructions

**Detailed description**

This PR replaces  #3283

* Search for QT6 by default
* Replace `-DCUTTER_QT6=ON` with `-DCUTTER_QT=5|6` 
* KF6SyntaxHighlighting
* Update build documentation 

**Test plan (required)**

* [x] CI green
* [x]  inspect the build config summary in CI logs
* [x] compare the package content with previous  CI builds (tested Windows zip and the main Appimage), 
* [ ] test that python plugins work using at least one of the CI packages

**Closing issues**

* Closes #3221

### CODE REVIEW

**Comment by karliss** [comment]
Some observations while testing the build on various distros:
* Ubuntu 24.04 lacks the QT6 version for the optional libraries like KFSyntaxHighlighter and pyside/shiboken - not  sure if that's a cause or consequences of many Ubuntu Qt  programs still being built against Qt5.
* No consequences on us, but the Ubuntu24.04 has 4 years old graphviz version. I understand that Ubuntu is far from being bleeding edge, but I didn't  expect it to be that old.
* On fedora shiboken6-devel package doesn't have a dependency on python3-devel or clang

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-    "cmake.sourceDirectory": "${workspaceFolder}"
+    "cmake.sourceDirectory": "${workspaceFolder}",
```
Comment (karliss) [inline .vscode/settings.json]: TODO: remove unintentional changes.

File: CMakeLists.txt [hunk 2]
```diff
@@ -145,22 +146,19 @@ if(CUTTER_ENABLE_PYTHON)
...
     else()
-        find_package(KF5SyntaxHighlighting REQUIRED)
+        find_package(KF${CUTTER_QT}SyntaxHighlighting REQUIRED)
         set(KSYNTAXHIGHLIGHTING_STATUS ON)
     endif()
 else()
-    if(CUTTER_ENABLE_KSYNTAXHIGHLIGHTING AND CUTTER_QT6)
-        message(WARNING "KSyntaxHighlighting has been disabled because not supported in QT6")
-    endif()
     set(KSYNTAXHIGHLIGHTING_STATUS OFF)
```
Comment (karliss) [inline CMakeLists.txt]: else branch is currently useless.

File: docs/source/building.rst [hunk 3]
```diff
@@ -61,22 +63,14 @@ On Debian-based Linux distributions, all of these essential packages can be inst
...
-   sudo apt install build-essential cmake meson libzip-dev zlib1g-dev qt5-default libqt5svg5-dev qttools5-dev qttools5-dev-tools
-
-.. note::
- On Debian 11 (bullseye) and higher or Ubuntu 22.04 (Jammy) and higher, replace ``qt5-default`` above with ``qtbase5-dev``.
+   sudo apt install build-essential cmake meson pkgconf libzip-dev zlib1g-dev qt6-base-dev qt6-tools-dev qt6-tools-dev-tools libqt6svg6-dev libqt6core5compat6-dev libqt6svgwidgets6 qt6-l10n-tools
 
 Depending on your configuration you'll might also need the following:
 
-::
+   ::
```
Comment (karliss) [inline docs/source/building.rst]: Unintendional changes.

**Comment by karliss** [comment]
Removed the CMAKE prefix path, section from windows instructions because with latest Qt and CMake on windows, CMake somehow find Qt even if it's not installed in default path as long as the Qt bin folder is in path. 

* So It seems like unless you have some very unusual setup it's unnecessary
* I couldn't test what's a suitable Qt subfolder for assigning within CMAKE_PREFIX_PATH, and the CMAKE module search rules are a bit tricky
* If someone manages to make a setup which doesn't work, we already mention CMAKE_PREFIX_PATH lower in the common troubleshooting section. The error message from CMake also talks about it.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -567,8 +561,8 @@ if(CUTTER_ENABLE_PYTHON)
     endif()
 endif()
 
-if(TARGET KF5::SyntaxHighlighting)
-    target_link_libraries(Cutter PRIVATE KF5::SyntaxHighlighting)
+if(TARGET KF{CUTTER_QT}::SyntaxHighlighting)
```
Comment (karliss) [inline src/CMakeLists.txt]: {} without dollar sign. Need to retest that it actually works.

**Review by wargio — APPROVED** [review]

**Review by XVilka — APPROVED** [review]
Nice, also CMake code simplified a bit

**Pull request merged**